### PR TITLE
🎨 The user can remove themselves from the project/workspace. :rotating_light: 

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- 🎨 The user can remove themselves from the project/workspace.

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]
it's cool, but a few tests might be a good idea to add.

File: services/web/server/src/simcore_service_webserver/projects/_groups_api.py [hunk 1]
```diff
@@ -139,13 +139,15 @@ async def delete_project_group(
...
 ) -> None:
-    await check_user_project_permission(
-        app,
-        project_id=project_id,
-        user_id=user_id,
-        product_name=product_name,
-        permission="delete",
-    )
+    user = await users_api.get_user(app, user_id=user_id)
+    if user["primary_gid"] != group_id:
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_groups_api.py:143]: Q: so here in reality you are deleting **the relation** between a project and a group, and not the latter, right?

File: services/web/server/src/simcore_service_webserver/projects/_groups_api.py [hunk 2]
```diff
@@ -139,13 +139,15 @@ async def delete_project_group(
...
+        await check_user_project_permission(
+            app,
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_groups_api.py:145]: Q: Not sure I fully understand why `check_user_project_permission` does not include the case in which the target `group_id` is a `primary_id`

**Review by GitHK — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/projects/_groups_api.py (hunk 1 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/_groups_api.py:143]: correct :+1:

File: services/web/server/src/simcore_service_webserver/projects/_groups_api.py (hunk 2 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/_groups_api.py:145]: In that case, it is the user's group, therefore he can delete himself from the project or workspace permissions even though he doesn't have write or delete permissions.

**Pull request merged**